### PR TITLE
IND-19421|Switch between endpoints to fetch the agent script

### PR DESCRIPTION
**Pull request by 1996sajal**
Agent script url was updated to https://<fqdn>/install/controller-agent as of 3.11. This MR adds support for older version of NGINX Controller.

### CODE REVIEW

File: amazon/Dockerfile [hunk 1]
```diff
@@ -55,7 +55,8 @@ RUN set -ex \
   # NGINX Javascript module needed for APIM
   && yum update && yum -y install nginx-plus nginx-plus-module-njs  \
   # Install Controller Agent
-  && curl -k -sS -L ${CONTROLLER_URL}/install/controller/ > install.sh \
+  && curl -k -sS -L ${CONTROLLER_URL}/ > install.sh \
```
Comment (azych) [inline amazon/Dockerfile:58]: I think the trailing `/` is not needed, so might be a good idea to remove it from here and other Dockerfiles

**Review by azych — APPROVED** [review]
Changes in the agent-layer entrypoints seems to be missing, other than that looks good

**Review by luk90 — APPROVED** [review]

**Comment by luk90** [comment]
https://github.com/nginxinc/docker-nginx-controller/blob/master/ubuntu/Dockerfile#L12
Please update the comment with new format of the value

File: amazon/Dockerfile (hunk 1 above)
Comment (1996sajal) [reply amazon/Dockerfile]: Resolved.

File: README.md [hunk 2]
```diff
@@ -30,15 +30,17 @@ Guidance around NGINX Plus is available [here](https://www.nginx.com/blog/deploy
...
 Note: When building NGINX Plus into a container, be sure to remove repository credentials from your container.
 
-### 1.1. NGINX Controller Agent Inside Docker Container
+### 1.1. Current Limitations
 
-The Controller Agent can be deployed in a Docker environment to monitor and/or configure NGINX processes inside Docker containers.
-The Controller Agent can collect most of the metrics.
+The following list summarizes the existing limitations of monitoring containers with NGINX Controller:
+
+- The Controller Agent can only monitor NGINX from inside the container. It is not currently possible to run the Controller Agent in a separate container and monitor the neighboring containers running NGINX.
```
Comment (brianehlert) [inline README.md:37]: We should not say that it is 'currently not possible' - currently implies that we intend to support this, which we do not.
Agent must be installed in / run in the same container as the N+ instance it is managing / monitoring - just as it does with a machine (must be installed along-side N+)

File: README.md [hunk 3]
```diff
@@ -30,15 +30,17 @@ Guidance around NGINX Plus is available [here](https://www.nginx.com/blog/deploy
 Dockerfiles contained in this repository are supported by and tested against NGINX Controller version 3.10 and later.
 Note: When building NGINX Plus into a container, be sure to remove repository credentials from your container.
 
-### 1.1. NGINX Controller Agent Inside Docker Container
+### 1.1. Current Limitations
```
Comment (brianehlert) [inline README.md:33]: These are requirements of agent, I would not frame them as limitations.
The requirements of agent in a container are not different than agent with N+ in a machine.
What is different in a container environment is the possible combinations that can be dreamed up.

File: README.md [hunk 4]
```diff
@@ -30,15 +30,17 @@ Guidance around NGINX Plus is available [here](https://www.nginx.com/blog/deploy
...
+The following list summarizes the existing limitations of monitoring containers with NGINX Controller:
+
+- The Controller Agent can only monitor NGINX from inside the container. It is not currently possible to run the Controller Agent in a separate container and monitor the neighboring containers running NGINX.
 
-The "agent-inside-the-container" is currently the only mode of operation. In other words, the Controller Agent should be running in the same container as the NGINX process being managed/monitored.
-For more information, please refer to our [Controller Dockerfile repository](https://github.com/nginxinc/docker-nginx-controller.git).
+### 1.2. NGINX Controller Agent inside Docker container
 
-### 1.2. Standalone Mode
+The Controller Agent can be deployed in a Docker environment to monitor and/or configure NGINX processes inside Docker containers. The Controller Agent can collect most of the metrics.
```
Comment (brianehlert) [inline README.md:41]: We should use the generic term 'container' or 'containers' - we are not opinionated with the management layer being used - in this case that is 'Docker'.  But is could be Openstack, K8s, mesos, etc.  So we should begin the pattern of not calling out hte management layer unless something is specific / limited to one particular management layer.

File: README.md [hunk 5]
```diff
@@ -30,15 +30,17 @@ Guidance around NGINX Plus is available [here](https://www.nginx.com/blog/deploy
...
+
+### 1.3. Standalone Mode
 
 By default, the Controller Agent will determine the OS hostname during installation using the `hostname -f` command. The hostname value will then be assigned to the `instance_name` key in the Controller Agent configuration file (`agent.conf`) and further used to generate a UUID, which together with an instance name provide a means of uniquely identifying the NGINX instance in NGINX Controller. When the Agent is run inside a container, the default hostname is a shortened Docker Container ID on the host. You can override the automatically assigned `instance_name` on runtime by setting the `ENV_CONTROLLER_INSTANCE_NAME` environment variable to the desired value.
```
Comment (brianehlert) [inline README.md:45]: We have a lot of detail here about how agent functions - detail that we don't document in the agent documentation.
I would simplify this by stating something along the line of: 
_hostname is used by default to identify the instances in NGINX Controller, this behavior can be overridden by using the run time environment variable ENV_CONTROLLER_INSTANCE_NAME.
If specifying this variable you need to ensure that this is unique for each container._

File: README.md [hunk 6]
```diff
@@ -30,15 +30,17 @@ Guidance around NGINX Plus is available [here](https://www.nginx.com/blog/deploy
...
+- The Controller Agent can only monitor NGINX from inside the container. It is not currently possible to run the Controller Agent in a separate container and monitor the neighboring containers running NGINX.
 
-The "agent-inside-the-container" is currently the only mode of operation. In other words, the Controller Agent should be running in the same container as the NGINX process being managed/monitored.
-For more information, please refer to our [Controller Dockerfile repository](https://github.com/nginxinc/docker-nginx-controller.git).
+### 1.2. NGINX Controller Agent inside Docker container
 
-### 1.2. Standalone Mode
+The Controller Agent can be deployed in a Docker environment to monitor and/or configure NGINX processes inside Docker containers. The Controller Agent can collect most of the metrics.
+
+### 1.3. Standalone Mode
```
Comment (brianehlert) [inline README.md:43]: This is about runtime - it should be merged with the runtime section.
With Controller 3.x all instances are 'stand-alone' - this is legacy language from the Amplify history that we need to remove.

Containers have build time guidance and run-time guidance.

File: README.md [hunk 7]
```diff
@@ -192,11 +192,11 @@ CONTAINER ID        IMAGE               COMMAND                  CREATED
 7d7b47ba4c72        nginx-agent       "/entrypoint.sh"         22 minutes ago      Exited (137) 19 seconds ago                       mynginx1
 ```
 
-## 3.0 Adding the Controller Agent during container run
+### 2.3. Adding the Controller Agent during container run
 
 An alternate way to handle the Controller Agent within containers is to include the necessary Controller Agent commands in the run command for the container. This way, you don't have to build the Controller Agent into your container before running.
```
Comment (brianehlert) [inline README.md:197]: What is happening here is the customer is beginning with an existing image and adding an additional layer that contains only the Controller Agent.
This is great for PoCs.  We recommend incorporating agent into the regular image build process for production.

**Comment by 1996sajal** [comment]
@brianehlert merged this.

**Pull request merged**